targets/TARGET_ublox/TARGET_R5030/serial_api.c

```diff
@@ -151,71 +171,107 @@ uint32_t zeroBitNumFromLsb(uint32_t data)
-     * data rate is 9600 because that's the only data rate that
-     * the 32 kHz (i.e. RTC) clock driving the LP UART can support.
-     *
-     * So, if the data rate is 9600, use the LP_UART/UART0
-     * combination, otherwise use UART0 for both Rx and Tx.  However,
-     * we don't know the data rate at this point so assume LP_UART
-     * (as this works at the default baud rate) and we can change
-     * our minds later.
-     *
-     * There is another serial port, UART1, which is normally used
-     * by the modem processor to send out debug.  We only initialise
-     * that here if the Tx pin is UART1_TX. */
-
-    /* Wait for the clock to be stable */
-    // while ((clock < CLK_FREQ_LOWTARGET) || (clock > CLK_FREQ_HIGHTARGET)) {
-        // clock = CLK_FREQ_DIG_CLKS;
-    // }
-
-    // if (tx == UART1_TX) {
-        // /* Use UART1 for Rx and Tx */
-        // obj->config = SERIAL_CONFIG_UART1_RX_UART1_TX;
-    // } else {
-        // /* Use LP_UART for Rx, UART0 for Tx */
-        // obj->config = SERIAL_CONFIG_UARTLP_RX_UART0_TX;
-    // }
-
-    // obj->rx_pin = rx;
-    // obj->tx_pin = tx;
-    // init_config(obj);
-
-    // /* TODO: set rx pin Pull mode ? */
-
-    // /* set default baud rate and format */
-    // serial_baud(obj, 9600);
-    // serial_format(obj, 8, ParityNone, 1);
-
-    // if (tx == UART0_TX) {
-        // /* The UART0 pins are the stdio pins */
-        // stdio_uart_inited = 1;
-        // stdio_uart = *obj;
-    // }
+{    
+    obj->rx_pin = rx;
+    obj->tx_pin = tx;
+        
+    if (tx == UART1_TX) {
+        serial_channel[0]=Uart1TXDxSIO_CHANNEL;
+        serial_channel[1]=Uart1RXDxSIO_CHANNEL;
+        config_pio_channel(serial_channel[0]);
+        config_pio_channel(serial_channel[1]);
+        obj->reg_base = app_ss_app_uart1;
+        obj->index = IRQ_UART_ID_1;
+        
+    } else if (tx == UART2_TX) {    
+        serial_channel[0]=Uart2TXDxSIO_CHANNEL;
+        serial_channel[1]=Uart2RXDxSIO_CHANNEL;
+        config_pio_channel(serial_channel[0]);
+        config_pio_channel(serial_channel[1]);
+        obj->reg_base = app_ss_app_uart2;
+        obj->index = IRQ_UART_ID_2;
+    }
+    else {
+        MBED_ASSERT(false);
+    }
+    
+    /* flush FIFOs and disable interrupts */
+    obj->reg_base->tffr = 0x0; 
+    obj->reg_base->rffr = 0x0; 
+    obj->reg_base->imrc = 0xFFFFFFF;
+    //obj->reg_base->imrs = 0x201; //enable tx shift register empty and rx not empty interrupts 
+    
+    serial_baud(obj, 9600);
+    serial_format(obj, 8, ParityNone, 1);
+    
+    if (tx == UART1_TX) {        
+        /* The UART1 pins are the stdio pins */
+        stdio_uart_inited = 1;
+        stdio_uart = *obj;
+    }        
+
 }
 
 void serial_free(serial_t *obj)
 {
-    // if (obj->tx_pin == UART0_TX) {
-        // stdio_uart_inited = 0;
-    // }
-
-    // serial_irq_ids[obj->index] = 0;
-
-    // /* Release the port HW */
-    // deinit_config(obj);
+    if (obj->tx_pin == UART1_TX) {
```

Comment:
Do we need check on UART2_TX ?

Reply:
Its not needed as stdio_uart_inited is set only for UART1.

Reply:
make sense